lib/google_contacts_api/contact.rb

```diff
@@ -137,32 +149,37 @@ def emails_full
     end
 
   private
-    def format_address(unformatted)
+
+    def format_email(unformatted)
       formatted = {}
-      formatted[:rel] = unformatted['rel'] ? unformatted['rel'].gsub('http://schemas.google.com/g/2005#', '') : 'work'
-      unformatted.delete 'rel'
-      unformatted.each do |key, value|
-        formatted[key.sub('gd$', '').underscore.to_sym] = value['$t']
-      end
+      rel = unformatted[:rel]
+      unformatted["primary"] ? formatted[:primary] = true : formatted[:primary] = false
+      formatted[:address] = unformatted["address"]
+      formatted[:rel] = rel.gsub('http://schemas.google.com/g/2005#', '')
       formatted
     end
 
-    def format_email_or_phone(unformatted)
+    def format_number(unformatted)
       formatted = {}
-      unformatted.each do |key, value|
-        formatted[key.underscore.to_sym] = value ? value.gsub('http://schemas.google.com/g/2005#', '') : value
-      end
-      formatted[:primary] = unformatted['primary'] ? unformatted['primary'] == 'true' : false
+      unformatted["primary"] ? formatted[:primary] = true : formatted[:primary] = false
```

Comment:
Erm... this is a poorly structured ternary. Have a think how this might be better structured